Source/Core/Resource.js

```diff
@@ -2059,59 +2059,65 @@ function loadWithHttpRequest(
   overrideMimeType
 ) {
   // Note: only the 'json' and 'text' responseTypes transforms the loaded buffer
-  /* eslint-disable no-undef */
-  const URL = require("url").parse(url);
-  const http = URL.protocol === "https:" ? require("https") : require("http");
-  const zlib = require("zlib");
-  /* eslint-enable no-undef */
-
-  const options = {
-    protocol: URL.protocol,
-    hostname: URL.hostname,
-    port: URL.port,
-    path: URL.path,
-    query: URL.query,
-    method: method,
-    headers: headers,
-  };
-
-  http
-    .request(options)
-    .on("response", function (res) {
-      if (res.statusCode < 200 || res.statusCode >= 300) {
-        deferred.reject(
-          new RequestErrorEvent(res.statusCode, res, res.headers)
-        );
-        return;
-      }
+  let URL;
+  let zlib;
+  Promise.all([import("url"), import("zlib")])
```

Comment:
Just to make sure - these imports are cached, right?

Reply:
This code should only ever be executed in environments where `XMLHttpRequest` is undefined, which is not in the browser but in NodeJS. In that case, the are dynamically imported from the `node_modules` directory.